### PR TITLE
add init-test to ray-core, bring back opencensus

### LINKED ISSUE #16: Missing package opencensus in ray-core?
Sorry for not using the big issue template, I think this error is quite easy to reproduce.

I created a new environment with 

```
mamba create -n test ray-core
```
Then on ```ray.init()```:

```
$ python
Python 3.9.5 (default, Jun  4 2021, 12:28:51)
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import ray
>>> ray.init()
```

I get a huge error message that finally tells me it tried to import opencensus and failed.

Installing opencensus with ```mamba install opencensus``` fixes the problem.

### PR DESCRIPTION
**Pull request by h-vetinari**
Add more tests, re-introduce `opencensus` as a requirement for `ray-core`.

Closes #16.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -116,6 +116,12 @@ outputs:
         - ray.runtime_context
         - ray.state
         - ray.worker
+      commands:
+        - python -c "import ray; ray.init()"
+        # init-code seemingly depends on platform or other ambient things;
+        # the following doesn't get triggered in CI, but was a problem in
+        # https://github.com/conda-forge/ray-packages-feedstock/issues/16
+        - python -c "import ray.metrics_agent"
```
Comment (vnlitvinov) [inline recipe/meta.yaml:124]: you can move that to upper part, somewhere around `- ray.worker` on line 118 - that section handles simple `import`s just fine

Comment (vnlitvinov) [reply recipe/meta.yaml]: I've moved that myself, hope you won't mind me touching your PR :)

**Review by vnlitvinov — APPROVED** [review]

File: recipe/meta.yaml (hunk 1 above)
Comment (h-vetinari) [reply recipe/meta.yaml]: No problem whatsoever, that's what co-maintainership is for!

**Comment by h-vetinari** [comment]
Thanks for running with this! :)

**Pull request merged**